drainer/collector_test.go

```diff
@@ -48,6 +49,59 @@ type dummyStore struct {
 
 type collectorSuite struct{}
 
+type dummyStorage struct {
+	kv.Storage
+}
+
+func (ds dummyStorage) Begin() (kv.Transaction, error) {
+	return nil, nil
+}
+
+func (ds dummyStorage) BeginWithStartTS(startTS uint64) (kv.Transaction, error) {
+	return nil, nil
+}
+
+func (ds dummyStorage) GetSnapshot(ver kv.Version) (kv.Snapshot, error) {
+	return nil, nil
+}
+
+func (ds dummyStorage) GetClient() kv.Client {
+	return nil
+}
+
+func (ds dummyStorage) Close() error {
+	return nil
+}
+
+func (ds dummyStorage) UUID() string {
+	return ""
+}
+
+func (ds dummyStorage) CurrentVersion() (kv.Version, error) {
```

Comment:
It seems only `CurrentVersion` is required in this case. You can try leaving the other dummy implementations out and see if it still works.